v3/otm_meshless.cpp

```diff
@@ -356,30 +358,30 @@ void otm_update_nodal_position(state& s) {
   auto const boundary_indices = s.boundaries;
   auto const boundary_to_prescribed_v = s.prescribed_v.cbegin();
   auto const boundary_to_prescribed_dof = s.prescribed_dof.cbegin();
-  for (auto boundary_index : boundary_indices) {
```

Comment:
I'm confused about how switching the loop structure causes errors in the boundary conditions. It seems to me that this loop is fully invertible.

Another option would be to keep the boundary conditions on the device, but you would need to hpc::copy them from host to device at initialization.

Reply:
@lxmota see my boundary condition PR. This issue should be fixed (please test) and working with CUDA build.